### DISCUSSION TITLE
想请教一个问题,关于多线路同步负载加速能否实现?

### DISCUSSION THREAD

**Discussion by linsan970809** [v2fly/v2ray-core#2451] [Q&A]
1.最近我们公司想弄外面的直播,但是直连的话经常卡断线,想问问V2RAY能不能解决这个问题?
2.硬环境如下:
       我们公司有三条宽带电信,联通和移动都是千兆的,上传实测50兆,我们在香港有一台50M的中转主机.
 3.目前的运行方式是:
      这三条宽带通过香港主机中转数据,每条线路在某一时段都是很流畅的,但是偶尔都有几分钟到十几分钟的PING值超高卡顿,换一条宽带就没事.比如电信直播中,突然不稳定了,手动切换到移动或联通马上就流畅了.我们想了很多办法,比如BBR,KCPTUN,mkcp等,都是这样.所以我就想了一个思路,不知道V2RAY能不能实现这种负载方式
4.如图:
        比如V2RAY同时用三条线路向香港服务器发送数据,服务端先收到哪一条线路的数据,就丢弃另外两条线路的数据包,往返都走这个思路,这样子只要有一条线路是稳定流程的,直播的数据就不会卡顿,除非三条宽带都出问题,我翻阅了V2RAY的文档,目前都是以轮询或者是随机,又或者是故障切换的方式进行的均衡负载,没有对速度要求高的负载方案.如果哪位大神有办法实现我的想法,请分享一下,不胜感激.

![20230406122337_1](https://user-images.githubusercontent.com/61045370/230271637-b712b8b2-9644-4b68-9ba1-6fd77b6d774e.jpg)

**Comment by mydogshitgold** [comment]
So you are like 3 or more public IPs assigned to your local host, with each IP corresponding to a different carrier? You can choose which public IP to use via `outbounds[].sendThrough` and V2Ray routing rules.

>The server receives the packet of which line first, and discards the packets of the other two lines.

I don't think V2Ray load balancing works in this way. It's connection oriented and does NOT send packets simultaneously through different public IPs and pick whichever one arrives first while ditching any others.

> No load schemes with high speed requirements.

If you mean "low latency", V2Ray has a "[leastping](https://github.com/v2fly/v2ray-core/blob/master/app/router/strategy_leastping.go)" strategy.

**Reply by linsan970809** [reply]
感谢你的回答,
        智能路由器可以支持同时接入多条宽带,然后根据不同的端口访问指定的IP地址来实现分流同时访问香港的服务器,leastping我也研究过,但是他对实时低延迟还是需要判断以后才能做出响应,也不符合一直低延迟的需求

**Reply by mydogshitgold** [reply]
>The smart router can support access to multiple broadband at the same time, and then access the specified IP address according to different ports to achieve splitting and access the server in Hong Kong at the same time

Whether to use port mapping or `sendThrough` is tangential to the topic at hand.


>I have also studied leastping, but it still needs to be judged before it can respond to low latency in real time, and it does not meet the need for low latency all the time.

No can do.

AFAIK V2Ray lacks the capability to concurrently send identical packets to multiple outbounds and subsequently select the winner of the race. You can whether wait for someone to implement this functionality, settle for the most comparable option which is `leastping`, or, in case that the instability of your connections is not attributed solely to the destination server, implement this functionality beneath V2Ray, akin to the methodology employed by Speedify.

**Reply by linsan970809** [reply]
非常感谢你的回复
